posts/2021-04-16-mp-context-propagration-21004.adoc

```diff
@@ -0,0 +1,387 @@
+----
+
+To enable this new function in your app, add the LDAP User Registry 3.0 feature to your `server.xml` file:
+
+[source, xml]
+----
+<featureManager>
+  <feature>ldapRegistry-3.0</feature>
+</featureManager>
+----
+
+For more information on `LdapRegistry`, see the link:https://openliberty.io/docs/latest/reference/feature/ldapRegistry-3.0.html[LDAP User Registry documentation].
+
+[#SSO]
+== Single Sign-On (SSO) LTPA and JWT support authentication filter
+
+With this new enhancement users can now use the authentication filter to select which HTTP servlet request should use link:{url-prefix}/docs/latest/reference/feature/appSecurity-3.0.html[LTPA] and link:{url-prefix}/docs/latest/reference/feature/jwtSso-1.0.html[JWT] for the SSO authentications.
+
+Users can configure an authentication filter to specify whether certain requests for protected resources are authenticated with LTPA. If the request meets the criteria that are specified in the authentication filter, then the request can authenticate with LTPA to access the protected resource. Conversely, if the request does not meet the criteria that are configured in the LTPA authentication filter, then the user is prompted to provide login credentials.
+
+[source, xml]
+----
+<ltpa keysFileName="yourLTPAKeysFileName.keys" keysPassword="keysPassword" expiration="120" authFilterRef="myAuthFilter"/>
+
+<authFilter id="myAuthFilter">
+         <requestUrl id="myRequestUrl" urlPattern="/SimpleServlet" matchType="contains"/>
+</authFilter>
+----
+
+In the example above, request has a LTPA cookie and URLs that contain the `/SimpleServlet` pattern are authenticated by LTPA SSO authentication. However, if the request has a LTPA cookie but the URLs do not contain the `/SimpleServlet` pattern are authenticated by other authentication mechanisms.
+
+
+Or users can configure an authentication filter to specify whether certain requests for protected resources are authenticated with JWT SSO. If the request meets the criteria that are specified in the authentication filter, then the request can authenticate with JWT to access the protected resource. Conversely, if the request does not meet the criteria that are configured in the JWT SSO authentication filter, then the user is prompted to provide login credentials.
+
+[source, xml]
+----
+<jwtSso cookieName="myjwt" jwtBuilderRef="myBuilder" authFilterRef="myAuthFilter"/>
+<authFilter id="myAuthFilter">
+         <requestUrl id="myRequestUrl" urlPattern="/SimpleServlet" matchType="notContain"/>
+</authFilter>
+----
+
+In the example above, same as for the LTPA authentication filter example: the request has JWT cookie and URLs that contain the `/SimpleServlet` pattern are authenticated by JWT SSO authentication. However, if the request has a JWT cookie but the URLs do not contain the `/SimpleServlet` pattern are authenticated by other authentication mechanisms.
+
+For more information see: 
```

Comment:
Not rendering correctly on draft site:
`+ * Open Liberty Documentation for Authentication filters * Open Liberty Documentation for Authentication Filter (AuthFilter)`

Reply:
Forgot to add an extra `+` inbetween

Reply:
Done now